File: leetcode/2956-find-common-elements-between-two-arrays/findIntersectionValues.test.ts
import cases from "./cases.json";
import findIntersectionValues from "./findIntersectionValues";

test("find common elements between two arrays", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    console.time("time");
    const result = findIntersectionValues(
      testCase.input.nums1,
      testCase.input.nums2
    );
    console.timeEnd("time");
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/2956-find-common-elements-between-two-arrays/findIntersectionValues_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"reflect"
	"testing"

	"github.com/edram/leetcode/util"
)

type FindIntersectionValuesCase struct {
	Input struct {
		Nums1 []int
		Nums2 []int
	}

	Output []int
}

func TestFindIntersectionValues(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]FindIntersectionValuesCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := findIntersectionValues(testCase.Input.Nums1, testCase.Input.Nums2)
		if !reflect.DeepEqual(result, testCase.Output) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
